pkg/operator/reports_test.go

```diff
@@ -92,3 +101,190 @@ func TestGetNextReportPeriod(t *testing.T) {
+			report: testhelpers.NewReport(testReportName, testNamespace, testQueryName, reportStart, reportEnd, v1alpha1.ReportStatus{
+				Conditions: []metering.ReportCondition{
+					*util.NewReportCondition(metering.ReportRunning, v1.ConditionFalse, util.ReportFinishedReason, "mock-msg"),
+				},
+			}, nil, false),
+			expectRunning: true,
+		},
+		{
+			name: "reportingEnd is unset returns false",
+			report: testhelpers.NewReport(testReportName, testNamespace, testQueryName, reportStart, nil, v1alpha1.ReportStatus{
+				Conditions: []metering.ReportCondition{
+					*util.NewReportCondition(metering.ReportRunning, v1.ConditionFalse, util.ReportFinishedReason, "mock-msg"),
+				},
+			}, schedule, false),
+			expectRunning: false,
+		},
+		{
+			name: "reportingEnd > lastReportTime returns false",
+			report: testhelpers.NewReport(testReportName, testNamespace, testQueryName, reportStart, reportEnd, v1alpha1.ReportStatus{
+				Conditions: []metering.ReportCondition{
+					*util.NewReportCondition(metering.ReportRunning, v1.ConditionFalse, util.ReportFinishedReason, "mock-msg"),
+				},
+				LastReportTime: &metav1.Time{Time: reportStart.AddDate(0, 0, 0)},
+			}, schedule, false),
+			expectRunning: false,
+		},
+		{
+			name: "reportingEnd < lastReportTime returns true",
+			report: testhelpers.NewReport(testReportName, testNamespace, testQueryName, reportStart, reportEnd, v1alpha1.ReportStatus{
+				Conditions: []metering.ReportCondition{
+					*util.NewReportCondition(metering.ReportRunning, v1.ConditionFalse, util.ReportFinishedReason, "mock-msg"),
+				},
+				LastReportTime: &metav1.Time{Time: reportStart.AddDate(0, 2, 0)},
+			}, schedule, false),
+			expectRunning: true,
+		},
+		{
+			name: "spec.Reason set to 'Scheduled' returns false",
+			report: testhelpers.NewReport(testReportName, testNamespace, testQueryName, reportStart, reportEnd, v1alpha1.ReportStatus{
+				Conditions: []metering.ReportCondition{
+					*util.NewReportCondition(metering.ReportRunning, v1.ConditionFalse, util.ScheduledReason, "mock-msg"),
+				},
+			}, schedule, false),
+			expectRunning: false,
+		},
+		{
+			name: "spec.Reason set to 'InvalidReport' returns false",
+			report: testhelpers.NewReport(testReportName, testNamespace, testQueryName, reportStart, reportEnd, v1alpha1.ReportStatus{
+				Conditions: []metering.ReportCondition{
+					*util.NewReportCondition(metering.ReportRunning, v1.ConditionFalse, util.InvalidReportReason, "mock-msg"),
+				},
+			}, schedule, false),
+			expectRunning: false,
+		},
+		{
+			name: "spec.Status set to v1.ConditionTrue returns false",
+			report: testhelpers.NewReport(testReportName, testNamespace, testQueryName, reportStart, reportEnd, v1alpha1.ReportStatus{
+				Conditions: []metering.ReportCondition{
+					*util.NewReportCondition(metering.ReportRunning, v1.ConditionTrue, util.ReportFinishedReason, "mock-msg"),
+				},
+			}, schedule, false),
+			expectRunning: false,
```

Comment:
Hmm I think the `reason` here needs to be something else, or we need another case, because the intent was that we have a test case where `expectRunning` is true, when there's a running condition.

Currently this test case is basically stating: "we don't expect it to be running, when Running is true with reason finished" which doesn't make much sense. In our actual code, this shouldn't ever happen (unless we have a bug of course).

Reason is never finished if the Running condition is true, so this is testing a case we don't currently expect to happen. So this test case might be fine, but it may also be good to figure out what should happen here (perhaps we should be panicing if this happened in our code, and our test should verify we panic).

Anyways, we need another test case where Running is true, and expectRunning is true. Use `ScheduledReason` or `RunImmediatelyReason` to handle the happy path where a report is not finished, and is just running as regular.

Reply:
Turns out, the problem aws expectRunning is actually `expectFinished` so this problem is a bit different, mostly a matter of what condition reason is used.